adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -378,132 +369,131 @@ class ADAMContext(val sc: SparkContext) extends Serializable with Logging {
       ContextUtil.getConfiguration(job))
     if (Metrics.isRecording) records.instrument() else records
 
-    applyPredicate(records.map(p => vcc.convertToAnnotation(p._2.get)), predicate)
+    records.map(p => vcc.convertToAnnotation(p._2.get))
   }
 
-  def loadParquetVariantAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+  def loadParquetVariantAnnotations(
     filePath: String,
-    predicate: Option[Class[U]] = None,
+    predicate: Option[FilterPredicate] = None,
     projection: Option[Schema] = None): RDD[DatabaseVariantAnnotation] = {
-    adamLoad[DatabaseVariantAnnotation, U](filePath, predicate, projection)
+    adamLoad[DatabaseVariantAnnotation](filePath, predicate, projection)
   }
 
-  def loadVariantAnnotations[U <: ADAMPredicate[DatabaseVariantAnnotation]](
+  def loadVariantAnnotations(
     filePath: String,
-    predicate: Option[Class[U]] = None,
+    predicate: Option[FilterPredicate] = None,
```

Comment:
Yeah; this is the discussion that @laserson raised on #609. Uri makes good arguments on the ticket for that issue. I disagreed at first, but have come around to agree with Uri's arguments.

As an aside, if you are going to fail if a predicate is provided and the format isn't VCF, there's not really any benefit in allowing people to provide a predicate to the automagical function. Essentially, the automagical function just exists as a shim of convenience, so it should be difficult to make the automagical loaders throw an error.